browser/modules/AboutNewTab.jsm

```diff
@@ -370,8 +401,12 @@ let AboutNewTab = {
    *          The tile index from which the action came from.
    */
   reportSitesAction: function(message) {
-    let sites = NewTabUtils.links.getLinks().slice(0, message.data.length);
-    DirectoryLinksProvider.reportSitesAction(sites, message.data.action, message.data.index);
+    // Convert sites to objects.
+    for (let i = 0; i < message.data.sites.length; i++) {
```

Comment:
(to be extra fancy: `[for (site of messages.data.sites) JSON.stringify(site)]` :smile_cat:, but JSHint might cry)